variant/tokenizers/polypeptide_truncation.py

```diff
@@ -0,0 +1,92 @@
+"""A module for Polypeptide Truncation Tokenization."""
+from typing import Optional
+from variant.schemas.token_response_schema import PolypeptideTruncationToken,\
+    TokenMatchType
+from .polypeptide_sequence_variant_base import PolypeptideSequenceVariantBase
+
+
+class PolypeptideTruncation(PolypeptideSequenceVariantBase):
+    """Class for tokenizing Protein Substitution."""
+
+    def match(self, input_string) -> Optional[PolypeptideTruncationToken]:
+        """Return a PolypeptideTruncationToken match if one exists.  # noqa: D202, E501
+
+        :param str input_string: The input string to match
+        :return: A PolypeptideTruncationToken if a match exists.
+            Otherwise, None.
+        """
+
+        input_string = str(input_string)
+        psub_parts = None
+        self.psub = {
+            'amino_acid': None,
+            'position': None,
+            'new_amino_acid': None
+        }
+
+        if input_string is None:
+            return None
+
+        if '.' in input_string:
+            split_whitespace = input_string.split()
+            if not input_string.startswith('p.'):
+                if not len(split_whitespace) == 2:
+                    return None
+            else:
+                p_count = input_string.count('p.')
+                if p_count == 1:
+                    psub_parts = self.splitter.split(input_string)
+            if len(split_whitespace) == 2:
+                psub_parts = split_whitespace
+        else:
+            psub_parts = self.splitter.split(input_string)
+
+        self._get_psub(psub_parts)
+
+        if all(self.psub.values()):
```

Comment:
assuming I was right about the comment above, it would apply here too (ignore otherwise)